### PR TITLE
add support for CRD, pods/log and pods/exec

### PR DESCRIPTION
**Pull request by wm775825**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks a lot~
/assign

**Comment by hzxuzhonghu** [comment]
Seems karmada bot send spam notifications to upstream k8s owners, should change that.

**Comment by RainbowMango** [comment]
> Seems karmada bot send spam notifications to upstream k8s owners, should change that.

really? Do you mean Karmada bot invited you to review it? 
We should cleanup the owner files, right?

File: staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go [hunk 1]
```diff
@@ -126,3 +132,50 @@ func NewDefaultConfig(prefix string, codec runtime.Codec) *Config {
...
+}
+
+func (t *TransportConfig) ClientConfig() (*tls.Config, error) {
+	if err := t.LoadTLSConfigurations(); err != nil {
+		return nil, err
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(t.TrustedCAData)
+	tlsConfig := &tls.Config{MinVersion: tls.VersionTLS12, RootCAs: caCertPool}
```
Comment (XiShanYongYe-Chang) [inline staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go:168]: Do we need to set the encryption suite when setting up tls1.2?

Comment (wm775825) [reply staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go:168]: it's just a copy from etcd config

File: pkg/kubelet/client/kubelet_client.go [hunk 2]
```diff
@@ -155,10 +157,12 @@ type NodeConnectionInfoGetter struct {
 	insecureSkipTLSVerifyTransport http.RoundTripper
 	// preferredAddressTypes specifies the preferred order to use to find a node address
 	preferredAddressTypes []v1.NodeAddressType
+
+	clusterTransportGetter func(string) (*url.URL, http.RoundTripper, error)
 }
 
 // NewNodeConnectionInfoGetter creates a new NodeConnectionInfoGetter.
-func NewNodeConnectionInfoGetter(nodes NodeGetter, config KubeletClientConfig) (ConnectionInfoGetter, error) {
+func NewNodeConnectionInfoGetter(nodes NodeGetter, config KubeletClientConfig, fleet *registry.FleetClientSet) (ConnectionInfoGetter, error) {
```
Comment (XiShanYongYe-Chang) [inline pkg/kubelet/client/kubelet_client.go:165]: Is this handling the proxy of Node resources?

Comment (wm775825) [reply pkg/kubelet/client/kubelet_client.go:165]: GetConnectionInfo depend on requestinfo in  ctx to get nodes

File: pkg/registry/apps/deployment/storage/storage.go [hunk 3]
```diff
@@ -403,15 +405,14 @@ func (i *scaleUpdatedObjectInfo) Preconditions() *metav1.Preconditions {
...
+	}
+
 	deployment, ok := oldObj.DeepCopyObject().(*apps.Deployment)
 	if !ok {
 		return nil, errors.NewBadRequest(fmt.Sprintf("expected existing object type to be Deployment, got %T", deployment))
 	}
-
-	if oldObj == nil {
-		return deployment, nil
-	}
```
Comment (XiShanYongYe-Chang) [inline pkg/registry/apps/deployment/storage/storage.go:413]: Why delete this logic?

File: pkg/registry/apps/deployment/storage/storage.go [hunk 4]
```diff
@@ -304,6 +304,7 @@ func (r *ScaleREST) Destroy() {
 
 // Get retrieves object from Scale storage.
 func (r *ScaleREST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	genericapirequest.ClearSubresourceInRequestInfo(ctx)
```
Comment (XiShanYongYe-Chang) [inline pkg/registry/apps/deployment/storage/storage.go:307]: Why clean the subresource in the request info?

File: pkg/registry/apps/deployment/storage/storage.go (hunk 3 above)
Comment (XiShanYongYe-Chang) [reply pkg/registry/apps/deployment/storage/storage.go:413]: Get it, it will generate a panic.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/cc @RainbowMango

**Pull request merged**